libs/net/reports/ReportXlsExport.cs

```diff
@@ -0,0 +1,543 @@
+using System.Text.Json;
+using NPOI.SS.UserModel;
+using NPOI.SS.Util;
+using NPOI.XSSF.UserModel;
+using TNO.API.Areas.Services.Models.ReportInstance;
+using TNO.DAL.Services;
+
+namespace TNO.Reports;
+
+class GroupResult
+{
+    public GroupResult (string name, string date, int count, double tone) {
+        this.Name = name;
+        this.Date = date;
+        this.Count = count;
+        this.Tone = tone;
+    }
+    public string Name {get; set;}
+    public string Date {get; set;}
+    public int Count {get; set;}
+    public double Tone {get; set;}
+}
+
+class HeadlineResult
+{
+    public HeadlineResult (double tone, string date, string headline, string type, string? source, string? series) {
+        this.Tone = tone;
+        this.Date = date;
+        this.Headline = headline;
+        this.Type = type;
+        this.Source = source;
+        this.Series = series;
+    }
+    public double Tone {get; set;}
+    public string Date {get; set;}
+    public string Headline {get; set;}
+    public string Type {get; set;}
+    public string? Source {get; set;}
+    public string? Series {get; set;}
+}
+
+class TonesCount
+{    
+    public TonesCount()
+    {
+        this.Neutral = 0;
+        this.Positive = 0;
+        this.Negative = 0;
+    }
+    public int Neutral {get; set;}
+    public int Positive {get; set;}
+    public int Negative {get; set;}
+}
+
+/// <summary>
+/// ReportXlsExport class, provides a way to export a report to XLS.
+/// </summary>
+public class ReportXlsExport
+{
+  #region Variables
+
+  private readonly IReportInstanceService _reportInstanceService;
+  private readonly XSSFWorkbook workbook;
+  private readonly JsonSerializerOptions _serializerOptions;
+  private readonly XSSFSheet sheet;
+  private readonly XSSFColor green = new();
+  private ICellStyle regularStyle;
+  private ICellStyle boldStyle;
+  #endregion
+
+  #region Constructors
+  /// <summary>
+  /// Creates a new instance of a Report object.
+  /// </summary>
+  public ReportXlsExport(IReportInstanceService reportInstanceService, JsonSerializerOptions serializerOptions)
+  {
+    this._reportInstanceService = reportInstanceService;
+    this._serializerOptions = serializerOptions;
+    workbook = new XSSFWorkbook();
+    green.ARGBHex = "008000";
+    sheet = (XSSFSheet)workbook.CreateSheet("CBRA");
+    this.regularStyle = CreateStyle(10, false, false, HorizontalAlignment.Right);
+    this.boldStyle = CreateStyle(10, true, false, HorizontalAlignment.Right);
+  }
+  #endregion
+
+  #region Methods
+        /// <summary>
+        /// Creates a new instance of a Report object.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns>XSSFWorkbook</returns>
+        public XSSFWorkbook GenerateReport(int id)
```

Comment:
This is a misleading method.  It doesn't generate a report, it generates an Excel document.  Also the `id` argument is not a `Report.Id`, it is a `ReportInstance.Id`.  We have other methods with the same name that actually generate a report.  We don't want to cause confusion.

Rename the method to `GenerateExcel`.  
Rename the argument to `instanceId`.